src/CounterInterceptor/ClientCounter.cs

```diff
@@ -0,0 +1,89 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using System.Numerics;
+
+public class ClientCounter
+{
+    private const string NumberOfWorkflowExecutions = "numOfWorkflowExec";
+    private const string NumberOfSignals = "numOfSignals";
+    private const string NumberOfQueries = "numOfQueries";
+    private static Dictionary<string, Dictionary<string, BigInteger?>> perWorkflowIdDictionary =
+        new Dictionary<string, Dictionary<string, BigInteger?>>();
+
+    public static string Info()
+    {
+        string result = string.Empty;
+        foreach (var item in perWorkflowIdDictionary)
```

Comment:
Use linq. Assuming you have the previously mentioned record, this is probably good enough: `string.Join("\n", perWorkflowIdDictionary.Select(kvp -> "** Workflow ID: {kvp.Key}, Counts: {kvp.Value}"))` or something like that

Reply:
Fixed